posthog/models/filters/mixins/simplify.py

```diff
@@ -54,6 +65,17 @@ def simplify(self: T, team: "Team", **kwargs) -> T:
 
         return result.with_data({**updated_entities, "properties": prop_group})
 
+    def _clear_excess_levels(self, prop: Union["PropertyGroup", "Property"]):
+        from posthog.models.property import PropertyGroup
+
+        if isinstance(prop, PropertyGroup):
+            if len(prop.values) == 1:
+                return self._clear_excess_levels(prop.values[0])
+            else:
+                prop.values = [self._clear_excess_levels(p) for p in prop.values]
```

Comment:
this is trickyyy. Say we have:

```
{
    type: "OR" ,
    values: [
        { type: OR, values: [ {type: AND, values: [prop1]}]},
        { type: AND, values: [prop2, prop3]}
    ]
}
```

The current simplification would lead to:

```
{
    type: "OR" ,
    values: [
        prop1,
        { type: AND, values: [prop2, prop3]}
    ]
}
```

which is invalid, which is why I initially moved off this route.

But, now question is if we can ever get into the first case?

Reply:
Let me add a test. I added the skip argument to handle this issue. When the parent level has multiple items in `values` it should skip elimination on the next level even if there's only 1 value in order to prevent type mismatch like you pointed out here